src/helper/client.cpp

```diff
@@ -475,39 +475,23 @@ bool client::handle_command(network::request_shutdown::request &command)
 
 bool client::run_client_init()
 {
-    try {
-        static constexpr auto client_init_timeout{
-            std::chrono::milliseconds{500}};
-        return handle_message<network::client_init>(
-            *this, *broker_, client_init_timeout);
-    } catch (const unexpected_command &) {
-        send_error_response(*broker_);
-    }
-
-    return false;
+    static constexpr auto client_init_timeout{std::chrono::milliseconds{500}};
+    return handle_message<network::client_init>(
+        *this, *broker_, client_init_timeout, false);
```

Comment:
Basically what happen is that `client_init` will handle unexpected messages differently from the rest. `Client_init` will disconnect client(according to existing tests added by you) while the rest of command won't. Since you asked to move the handling of unexpected commands to handle_message, somehow I need to let it know what to do when `unexpected_command` exception is thrown

Reply:
I completely read this the other way around for some reason, I guess that's fine then.